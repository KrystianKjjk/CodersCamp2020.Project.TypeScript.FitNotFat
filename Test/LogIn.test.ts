import loginForm from '../Src/UIComponents/LogIn/LogIn';

const loginBtnLogic = jest.fn();
const username = 'User7';

describe(('tests for Log In Form Component'),()=>{
    
    beforeEach(() => {
        loginBtnLogic.mockClear();
    })

    test('render correctly', () => {
       const form = loginForm(loginBtnLogic);
       expect(form).toMatchSnapshot();
    })

    test("call mock function with login button and username", () => {
        const form = loginForm(loginBtnLogic);
        const nameInput = form.querySelector('input.name') as HTMLInputElement;
        const loginButton = form.querySelector('button');
        nameInput.value = username;
        loginButton.click();
        expect(loginBtnLogic).toBeCalledTimes(1);
        expect(loginBtnLogic).toBeCalledWith(loginButton, username);
    })
})