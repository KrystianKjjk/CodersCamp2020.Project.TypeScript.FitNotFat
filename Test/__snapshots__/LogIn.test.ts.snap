// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Log In Form Component render correctly 1`] = `
<div
  class="main-container"
>
  <p
    class="main-title"
  >
    Log In
  </p>
  <input
    class="name"
    placeholder="Nickname"
    required=""
    type="text"
  />
  <div
    class="container-button"
  >
    <button
      class="button"
      style="background-color: rgb(218, 27, 54); color: white;"
    />
  </div>
</div>
`;
